### PR TITLE
New Relic Integration in Autotrace

### PR DESCRIPTION
**Pull request by StaticallyTypedAnxiety**
This adds new relic integration in the autotrace

### CODE REVIEW

File: crates/tracing_macros/src/autotrace.rs [hunk 1]
```diff
@@ -1,13 +1,39 @@
...
+use std::env;
 use syn::{Attribute, ImplItemMethod, ItemFn};
 
 const DEBUG_OUTPUT: bool = false;
 
 #[derive(Default)]
-pub(crate) struct Autotrace {}
+pub(crate) struct Autotrace {
+    app_name: String,
+    license_key: Option<String>,
```
Comment (struktured) [inline crates/tracing_macros/src/autotrace.rs]: ??? Are we paying for this software?

File: crates/tracing_macros/src/autotrace.rs [hunk 2]
```diff
@@ -21,16 +47,53 @@ impl Autotrace {
...
+                            #block
+                        }
+                    }
+                    else
+                    {
+                        #block
+                    }
+                }}
+            })
+            .unwrap_or_else(|| {
```
Comment (struktured) [inline crates/tracing_macros/src/autotrace.rs]: So it falls back to the previous autotracing implementation- interesting. Is there any reason to do both?

**Review by struktured — APPROVED** [review]
Don't know much details about `Relic` but LGTM beyond some clarifying questions.

File: crates/tracing_macros/src/autotrace.rs (hunk 1 above)
Comment (StaticallyTypedAnxiety) [reply crates/tracing_macros/src/autotrace.rs]: We are free tier at this point but I am not so sure what it will be like in the future

File: crates/tracing_macros/src/autotrace.rs (hunk 2 above)
Comment (StaticallyTypedAnxiety) [reply crates/tracing_macros/src/autotrace.rs]: If something happens ever happens when contacting the newrelic external services or if it is not configured to use new_relic I would like it to behave as normal

**Review by maackle — CHANGES_REQUESTED** [review]
@AshantiMutinta I think autotrace for newrelic has to at least be a separate macro from the autotrace for generating spans. First of all, they are two completely different meanings of the word "tracing". Second, we want to use the existing form of autotrace much more surgically than the new relic kind. 

The existing autotrace is a convenience for creating a shotgun-style approach to automatically creating spans when functions are called, but in real use a more fine-grained approach is required. The new relic autotracing would probably be applied to every single function, but we don't want that for spans, because functions that receive a new span from some other thread need to handle it more manually.

Basically we need all four combinations of tracing to be possible: 

1. no tracing
2. newrelic but not spans
3. spans but not newrelic
4. both newrelic and spans

And that's not possible with one macro unless we add extra parameters, but then I don't see the point in having one instead of two. So, I think this should be a separate macro.

**Review by maackle — APPROVED** [review]
Thanks, looks good now!
